checkers_bot_tournament/board.py

```diff
@@ -1,21 +1,88 @@
+import re
+from typing import Optional, Tuple
+from enum import auto, Enum
+
 from checkers_bot_tournament.move import Move
 from checkers_bot_tournament.piece import Piece, Colour
 
-from typing import Optional, Tuple
+Grid = list[list[Optional[Piece]]]
+
+
+class BoardState(Enum):
```

Comment:
Tbh is an Enum really needed here

Reply:
Anyway better to call it BoardStart or something like that

Reply:
Yeah ok I'll rename it BoardStart. I made it an enum bc I'm gonna add more later, and it's a way of initialising a board without having to play out a set of moves to get there (i.e. with importing a pdn), e.g. playing a 2v2 king endgame or something very different from a normal board start.

Reply:
Yeah agree with your vision there. I was thinking if you're gonna have to enumerate the different options, how about explicitly defining them as subclasses:

```python
class BaseBoardStartBuilder(ABC):
    def build(self, size: int) -> Grid:
        ...
class DefaultBoardStartBuilder(BaseBoardStartBuilder):
    def build(self, size: int) -> Grid:
        # do default
class FenBoardStartBuilder(BaseBoardStartBuilder):
    def build(self, size: int, fen: 'str') -> Grid:
        # from custom fen position

class Board:
    def __init__(self, board_start_builder: BoardStartBuilder, size: int = 8):
        # instead of self.initialise_pieces()
        self.grid = board_start_builder.build()
def main():
    # match case here
    # Endgame mode selected, so use FenBoardStartBuilder
    board_start = FenBoardStartBuilder()
    
    new_board = Board(board_start)
```

Reply:
Yeah sure that sounds good, I'll make a new file for BoardStateBuilder classes then.